migration.go

```diff
@@ -0,0 +1,22 @@
+package config
+
+import "github.com/libp2p/go-libp2p-core/peer"
+
+const DefaultMigrationKeep = "cache"
+
+var DefaultMigrationDownloadSources = []string{"HTTPS", "IPFS"}
+
+// Migration configures how migrations are downloaded and if the downloads are
+// added to IPFS locally
+type Migration struct {
+	// Sources in order of preference where "HTTPS" means our gateways and
+	// "IPFS" means over IPFS. Any other values are interpretes as hostnames
+	// for custom gateways. An empty list means "do the default thing"
+	DownloadSources []string
```

Comment:
@Stebalien Confirming there's no reason to have a "do nothing" option here? I don't think so since the migrations are required for go-ipfs to run and the alternative would be to error.

Reply:
@lidel It may look like it should be `Transports`, but this section also allows users to enter their own gateway addresses. So, the "IPFS" and "HTTP" keywords can be seen as selecting a set of peers or gateways that are associated with those protocols.

@Stebalien Thoughts?

Reply:
@aschmahmann We could add a "NONE" in the future, but I'd like to keep the "default" as nothing for now at least (simpler).

@lidel Transport is not exactly accurate because we _do_ support specifying a specific gateway, IIRC.

Reply:
> but I'd like to keep the "default" as nothing for now at least (simpler).

I'd like us to be able to specify the defaults in code and change them. In particular, per your earlier suggestion we can prefer dist.ipfs.io HTTP to IPFS in the next release of go-ipfs and then easily switch to preferring IPFS in a subsequent release. If `[]` means do nothing then IIUC we'll have to put values into the config file to be useful at all which means changing them is something a user will have to do.

Reply:
Currently `[]` Means use default behavior:
https://github.com/ipfs/go-ipfs/blob/feat/migration-ipfs-download/cmd/ipfs/daemon.go#L315-L316

I am not about the use case for `NONE`.  If a user wants to avoid downloading migrations automatically, then they can download them manually, before starting the daemon, and put them into PATH.  The daemon will look in PATH to see if the migrations are already present before downloading them.

Reply:
Yeah, that's reasonable. Some day, we'll get around to refactoring our config design to make this suck less.